### PR TITLE
Implemented Stripe for Membership fee payments

### PR DESCRIPTION
**Pull request by tobyL05**
TLDR:
- Onboarding process now requires users to pay the membership fee.
- The credentials are on test mode right now so we're not actually charging anyone yet. Also, the protocol needs to be https to use apple pay or something.
- When the user is onboarded successfully, the user and the transaction they made gets added to the firestore

Testing:
- Add a new field to your frontend .env called VITE_STRIPE_KEY. You can find the key on the Stripe developer dashboard under API Keys. It should start with "pk_test_..."
- Test with my backend branch
- When a new user continues with Google, the onboarding process should be the same but there should be a payments page at the end. When the payment is successful, a payment success page will be shown and the user can press a button to go to the dashboard.

### CODE REVIEW

**Review by Hgjnnf — CHANGES_REQUESTED** [review]
Works pretty well! The payment works (both successful and unsuccessful payment), user is added to the database upon successful payment, and the ui looks mostly correct (doesn't match Figma but we can move that to the future). 

Need to uncomment the necessary parts.

So far:
- Payment does not lead to PaymentSuccess
- Onboarding form is not conditionally rendered

File: src/components/Payment/PaymentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          return;
+        }
+    
+        setIsLoading(true);
+
+        const { error, paymentIntent } = await stripe.confirmPayment({
+          elements,
+          confirmParams: {
+            // Make sure to change this to your payment completion page
+            return_url: "http://localhost:5173/dashboard", // make a payment success page
```
Comment (Hgjnnf) [inline src/components/Payment/PaymentForm.tsx:67]: This could be an env variable where the base url can be switched to the actual url in production.

Comment (Hgjnnf) [reply src/components/Payment/PaymentForm.tsx:67]: Okay, sounds good

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -36,19 +36,24 @@ export default function Login() {
           idToken: idToken
         })
       })
+
+
       if (login.ok) {
         // User exists so go to /dashboard
         localStorage.setItem('member_id', user.uid);
         localStorage.setItem('member_name', displayName || "guest");
```
Comment (tobyL05) [inline src/pages/Login/Login.tsx:44]: I don't think we need this anymore right? Since we should be using `auth.currentUser`

Comment (Hgjnnf) [reply src/pages/Login/Login.tsx]: yea we can remove this

**Review by Hgjnnf — APPROVED** [review]
Functionality works! Let's squash and merge this
